### PR TITLE
e-mail to email

### PR DESCRIPTION
**Pull request by OnkarRuikar**
Completes #23049

This PR
- updates e-mail in REVIEWING.md file. Some other fixes
- updates a broken redirect
- in writing guides page uses HTML code for hyphen so that linter won't flag the `e-mail` in the example

### CODE REVIEW

File: files/en-us/web/api/navigator/webdriver/index.md [hunk 1]
```diff
@@ -17,7 +17,7 @@ The **`webdriver`** read-only property
 of the {{domxref("navigator")}} interface indicates whether the user agent is
 controlled by automation.
 
-It defines a standard way for co-operating user agents to inform the document that it
+It defines a standard way for cooperating user agents to inform the document that it
```
Comment (wbamberg) [inline files/en-us/web/api/navigator/webdriver/index.md:20]: Doesn't this fall foul of the rule: "Hyphenated compounds should be used when the last letter of the prefix is a vowel and is the same as the first letter of the root." ?

File: files/en-us/mdn/writing_guidelines/writing_style_guide/index.md [hunk 2]
```diff
@@ -334,7 +334,7 @@ The list below describes some of the common situations where we need to be aware
 Hyphenated compounds should be used when the last letter of the prefix is a vowel and is the same as the first letter of the root.
 
 - **Correct**: email, re-elect, co-op
-- **Incorrect**: e-mail, reelect, coop
+- **Incorrect**: e&#45;mail, reelect, coop
```
Comment (wbamberg) [inline files/en-us/mdn/writing_guidelines/writing_style_guide/index.md:337]: FWIW it is odd including "e-mail" in here, unless the idea is that _all_ other compounds should omit the hyphen, (a) in which case it should say that, and (b) I'm not sure I agree with.

Comment (OnkarRuikar) [reply files/en-us/mdn/writing_guidelines/writing_style_guide/index.md]: I've removed the email from the example.

Comment (Josh-Cena) [reply files/en-us/mdn/writing_guidelines/writing_style_guide/index.md]: @wbamberg "E-mail" provides the example where the prefix is _not_ the same letter as the root and therefore does not get the hyphen. It occurs to me that this example proves the rule is both necessary and sufficient.

File: files/en-us/web/api/navigator/webdriver/index.md (hunk 1 above)
Comment (OnkarRuikar) [reply files/en-us/web/api/navigator/webdriver/index.md:20]: > Diaresis for the win!
> - Naïve

I don't think we allow those: https://github.com/mdn/content/pull/16202#issuecomment-1131664414

File: files/en-us/mdn/writing_guidelines/writing_style_guide/index.md (hunk 2 above)
Comment (wbamberg) [reply files/en-us/mdn/writing_guidelines/writing_style_guide/index.md]: I understood that. In that case the text should say:

> Hyphenated compounds should be used only when the last letter of the prefix is a vowel and is the same as the first letter of the root.

or

> Compound words should be hyphenated only when the last letter of the prefix is a vowel and is the same as the first letter of the root.

...which was what my (a) was supposed to mean. The example is just an example, the rule should be in the text.

File: files/en-us/web/api/navigator/webdriver/index.md (hunk 1 above)
Comment (OnkarRuikar) [reply files/en-us/web/api/navigator/webdriver/index.md]: I'v reverted this change. Do not want to block main e-mail related changes.

File: files/en-us/mdn/writing_guidelines/writing_style_guide/index.md (hunk 2 above)
Comment (OnkarRuikar) [reply files/en-us/mdn/writing_guidelines/writing_style_guide/index.md]: Used second suggestion from @wbamberg.

**Review by wbamberg — APPROVED** [review]
👍 thank you!

**Pull request merged**